src/Stratis.Bitcoin/Consensus/IConsensusRuleEngine.cs

```diff
@@ -60,19 +61,25 @@ public interface IConsensusRuleEngine : IDisposable
         Task<RewindState> RewindAsync();
 
         /// <summary>Execute header validation rules.</summary>
-        /// <param name="validationContext">The validation context.</param>
-        void HeaderValidation(ValidationContext validationContext);
+        /// <param name="header">Header to validate.</param>
+        /// <returns>Context that contains validation result related information.</returns>
+        ValidationContext HeaderValidation(ChainedHeader header);
 
         /// <summary>Execute integrity validation rules.</summary>
-        /// <param name="validationContext">The validation context.</param>
-        void IntegrityValidation(ValidationContext validationContext);
+        /// <param name="header">The chained header that is going to be validated.</param>
```

Comment:
The 'chained header' here, but on the 'Header' above on line 64.  Think they need to be consistent.

Reply:
fixed